tests/utils.go

```diff
@@ -2475,11 +2486,89 @@ func GetPolicyMatchedToVmi(name string, vmi *v1.VirtualMachineInstance, namespac
 	}
 
 	applyLabels(policy.Spec.Selectors.VirtualMachineInstanceSelector, vmi.Labels, matchingVmiLabels)
-	applyLabels(policy.Spec.Selectors.NamespaceSelector, namespace.Labels, matchingNSLabels)
+	applyLabels(policy.Spec.Selectors.NamespaceSelector, namespaceLabels, matchingNSLabels)
+
+	if namespace != nil {
+		namespace.Labels = namespaceLabels
+	}
 
 	return policy
 }
 
+// GetPolicyMatchedToVmi mutates the given vmi parameter by adding labels to it. Therefore, it's recommended
+// to use this function before creating the vmi. Otherwise, its labels need to be updated.
+func GetPolicyMatchedToVmi(vmi *v1.VirtualMachineInstance) *migrationsv1.MigrationPolicy {
+	return GetPolicyMatchedToVmiWithNsAndVmiLabels(vmi, nil, 1, 0)
+}
+
+// ApplyPolicyToVmi creates a new migration policy with the provided policySpec and updates / creates the VMI so it would
+// be matched to this policy using a single vmi-level label.
+func ApplyPolicyToVmi(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, policySpec migrationsv1.MigrationPolicySpec) (*v1.VirtualMachineInstance, *migrationsv1.MigrationPolicy) {
```

Comment:
So what confuses me a little bit. Can't you just create in the tests in question the VMIs with the right label, but just not create the policy yet, being able to avoid all patch operations entirely by just creating and deleting the policy in time? That would probably simplify all this and would make most of this helper stuff unnecessary.

Would that be possible?

Reply:
I wanted to provide a way of limiting a running vmi with a certain bandwidth, but on a second thought I think you're right. The helper function became more complex than I intended with no real use case in the tests.

Changed :+1: